File: mindy/week1/L-1940.py
# 투포인터 문제
# 투포인터 문제의 경우 보통 정렬해서 풀이

import sys

input = sys.stdin.readline

n = int(input().strip())
m = int(input().strip())

k = list(map(int,input().strip().split()))
cnt =0 
k.sort()

left, right = 0,len(k)-1

while left<right:
    sum = k[left]+k[right]
    if sum < m:
        left +=1
    elif sum > m:
        right -=1
    else:
        cnt+=1
        left+=1
        right -=1
 
print(cnt)               
        

File: leo/week1/L-1940.js
/**
 *  제목 : 주몽 투포인터
 *  입력 : 15000
 *  제한사항 : 2중 for문으로 해결불가
 *  시간 제한 : O 투포인터 사용하면 nlogn
 */

// Run by Node.js
const readline = require("readline");

(async () => {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  let input = [];

  rl.on("line", (line) => {
    input.push(line.trim());
  });

  rl.on("close", () => {
    const N = parseInt(input[0]);
    const M = parseInt(input[1]);
    let materials = input[2].split(" ").map(Number);

    materials.sort((a, b) => a - b);

    let count = 0;
    let left = 0;
    let right = N - 1;

    while (left < right) {
      const sum = materials[left] + materials[right];

      if (sum === M) {
        count++;
        left++;
        right--;
      } else if (sum < M) {
        left++;
      } else {
        right--;
      }
    }

    console.log(count);

    process.exit();
  });
})();
